solvers/mathematical_program_result.h

```diff
@@ -218,6 +231,51 @@ class MathematicalProgramResult final {
     return value;
   }
 
+  /**
+   * Gets the dual solution associated with a constraint.
+   *
+   * We interpret the dual variable value as the "shadow price" of the original
+   * problem. Namely if we change the constraint bound by one unit (each unit is
+   * infinitesimally small), the change of the optimal cost is the value of the
+   * dual solution. Mathematically dual_solution = ∂optimal_cost / ∂bound.
+   *
+   * For a linear equality constraint Ax=b where b ∈ ℝⁿ, the dual variables has
```

Comment:
Very minor: just to avoid the "dual variables has" maybe "the vector of dual variables has"?